numba/np/arraymath.py

```diff
@@ -809,110 +809,65 @@ def flat_all(a):
     return flat_all
 
 
-@register_jitable
-def _allclose_scalars(a_v, b_v, rtol=1e-05, atol=1e-08, equal_nan=False):
-    a_v_isnan = np.isnan(a_v)
-    b_v_isnan = np.isnan(b_v)
-
-    # only one of the values is NaN and the
-    # other is not.
-    if ( (not a_v_isnan and b_v_isnan) or
-            (a_v_isnan and not b_v_isnan) ):
+@overload(np.any)
+@overload_method(types.Array, "any")
+def np_any(a):
+    def flat_any(a):
+        for v in np.nditer(a):
+            if v.item():
+                return True
         return False
 
-    # either both of the values are NaN
-    # or both are numbers
-    if a_v_isnan and b_v_isnan:
-        if not equal_nan:
-            return False
-    else:
-        if np.isinf(a_v) or np.isinf(b_v):
-            return a_v == b_v
-
-        if np.abs(a_v - b_v) > atol + rtol * np.abs(b_v * 1.0):
-            return False
+    return flat_any
 
-    return True
 
+@register_jitable
+def _isclose_scalars(a, b, rtol, atol, equal_nan):
+    a_isnan = np.isnan(a)
+    b_isnan = np.isnan(b)
 
-@overload(np.allclose)
-@overload_method(types.Array, "allclose")
-def np_allclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
+    if a_isnan != b_isnan:
+        return False
 
-    if not type_can_asarray(a):
-        raise TypeError('The first argument "a" must be array-like')
-
-    if not type_can_asarray(b):
-        raise TypeError('The second argument "b" must be array-like')
-
-    if not isinstance(rtol, types.Float):
-        raise TypeError('The third argument "rtol" must be a '
-                        'floating point')
-
-    if not isinstance(atol, types.Float):
-        raise TypingError('The fourth argument "atol" must be a '
-                          'floating point')
-
-    if not isinstance(equal_nan, types.Boolean):
-        raise TypeError('The fifth argument "equal_nan" must be a '
-                        'boolean')
```

Comment:
When `np.allclose` is used with unsupported types, these exceptions provide a useful error message. Could we keep them?